backend/functions/tests/flashcard.test.js

```diff
@@ -32,115 +36,255 @@ describe("addFlashcardSet", () => {
-    await expect(addFlashcardSet(data, context)).rejects.toEqual(
-        new Error(
-            "The \"category\" field must be a string between 1-30 characters.",
-        ),
-    );
+    test("throws error if cards fail pre-condition", async () => {
+      const addFlashcardSet = fft.wrap(myFunctions.addFlashcardSet);
+      const data = {
+        title: "title",
+        category: "category",
+        cards: [{}],
+      };
+      const context = {auth: {uid: user.uid}};
+      const cases = [
+        [],
+        [{}],
+        [{question: ""}],
+        [{question: "", answer: ""}],
+        [{question: "question", answer: ""}],
+        [{question: "", answer: "answer"}],
+      ];
+
+      for (const testVal of cases) {
+        data.cards = testVal;
+        await expect(addFlashcardSet(data, context)).rejects.toEqual(
+            new Error(
+                "The \"cards\" field must be a non-empty array of objects with" +
+                " keys \"question\" and \"answer\" with non-empty string values.",
+            ),
+        );
+      }
+    });
   });
+  functions.logger.info("Test addFlashcardSet was succesfull");
 
-  test("throws error if cards fail pre-condition", async () => {
-    const addFlashcardSet = fft.wrap(myFunctions.addFlashcardSet);
-    const data = {
-      title: "title",
-      category: "category",
-      cards: [{}],
-    };
-    const context = {auth: {uid: user.uid}};
-    const cases = [
-      [],
-      [{}],
-      [{question: ""}],
-      [{question: "", answer: ""}],
-      [{question: "question", answer: ""}],
-      [{question: "", answer: "answer"}],
-    ];
-
-    for (const testVal of cases) {
-      data.cards = testVal;
-      await expect(addFlashcardSet(data, context)).rejects.toEqual(
+  //* **************** Test updateFlashcardSet ******************* */
+  describe("updateFlashcardSet", () => {
+    test("update doc in /flascards and /users", async () => {
+      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
+      const data = {
+        title: "Updated Flaschard Set Title",
+        category: "Updated category",
+        cards: [
+          {question: "Updated Question 1", answer: "Updated Answer 1"},
+          {question: "Updated Question 2", answer: "Updated Answer 2"},
+          {question: "New Question", answer: "New Answer"},
+        ],
+      };
+
+      // Create a test flashcard set
+      const addFlashcardSet = fft.wrap(myFunctions.addFlashcardSet);
+      const addData = {
+        title: "Test Flashcard Set",
+        category: "Test Category",
+        cards: [
+          {question: "Test Question 1", answer: "Test Answer 1"},
+          {question: "Test Question 2", answer: "Test Answer 2"},
+        ],
+      };
+
+      const context = {auth: {uid: user.uid}};
+
+      const addRes = await addFlashcardSet(addData, context);
+      docsToDelete.push(addRes.flashcardId);
+
+      // Validate reference to flashcard set is on "user" document
```

Comment:
Maybe can remove this comment `// Validate reference to flashcard set is on "user" document` since this isn't done in this test

Reply:
Or we might add a test to do so to make sure that in the user document, it updates the reference data (move this logic to after validating the response after updating)